tools/c7n_gcp/c7n_gcp/resources/compute.py

```diff
@@ -98,6 +98,49 @@ class Delete(InstanceAction):
         '.*?/projects/(.*?)/zones/(.*?)/instances/(.*)')
 
 
+@Instance.action_registry.register('detach-disks')
+class DetachDisks(MethodAction):
+    """
+    `Detaches <https://cloud.google.com/compute/docs/reference/rest/v1/instances/detachDisk>`_
+    all disks from instance. The action does not specify any parameters.
+
+    It may be useful to be used before deleting instances (to not delete disks that are set to auto delete)
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: gcp-instance-detach-disks
+            resource: gcp.instance
+            filters:
+              - type: value
+                key: name
+                value: instance-template-to-detahc
+            actions:
+              - type: detach-disks
+    """
+    schema = type_schema('detach-disks')
+    
+    def validate(self):
```

Comment:
you can skip the validate if its empty, also the validate signature changed to not require return of self anymore.

Reply:
I need to overwrite validate from `core`.